src/components/UserClaims/UserClaims.tsx

```diff
@@ -0,0 +1,299 @@
+        }
+      }
+    })
+  }, [])
+
+  return (
+    <div className="flex flex-col items-center w-full">
+      <div
+        className={`bg-white w-full lg:w-1/3 rounded-lg ${
+          moderator ? 'md:shadow-[0_0_0_8px_rgb(255,237,212)]' : 'shadow-md'
+        } transition flex text-sm flex-col`}
+      >
+        <div className="flex-grow border-solid h-80 pb-5 overflow-auto flex flex-col justify-end px-5">
+          {messages.map((m) => (
+            <Message
+              message={m}
+              local={m.author === author}
+              deleteMessage={deleteMessage}
+              moderator={moderator}
+              key={m.id}
+            />
+          ))}
+        </div>
+        <div className="pb-3 px-5 border-b flex justify-between">
+          <input
+            type="text"
+            disabled={loading}
+            className="bg-slate-100 rounded-full py-2 px-3 flex-grow mr-2"
+            placeholder="Type something..."
+            value={draft}
+            onChange={(e) => setDraft(e.target.value)}
+            onKeyDown={(e) => e.key === 'Enter' && sendMessage()}
+          />
+          <button
+            onClick={sendMessage}
+            className="bg-slate-800 text-white rounded-full p-2 hover:bg-slate-900 focus:bg-slate-700 border border-transparent disabled:bg-slate-200 focus:border-[rgba(14,165,233,0.3)]"
+            disabled={loading}
+          >
+            <PaperAirplaneIcon className="w-5 h-5 rotate-90" />
+          </button>
+        </div>
+        <PrivilegeBar
+          moderator={moderator}
+          loading={loading}
+          onToggle={switchMode}
+        />
+      </div>
+      <div className="p-6 w-full lg:w-1/3 mx-auto rounded-lg bg-purple-100 p-4 mt-6 flex">
+        <LightBulbIcon className="w-[5em] text-purple-600" />
+        <p className="text-purple-900 text-center flex-grow">
+          Send messages from one or more windows. Toggle between roles to delete
+          messages sent by you or by anyone else in the chat.
+        </p>
+      </div>
+    </div>
+  )
+}
+
+function PrivilegeBar({
+  moderator,
+  loading,
+  onToggle,
+}: {
+  moderator: boolean
+  loading: boolean
+  onToggle: () => void
+}) {
+  let iconColour = 'text-slate-500'
+  let iconBackground = 'bg-slate-100'
+  let Icon = RefreshIcon
+  let titleText = <>Switching roles</>
+  let subtitleText = <>Loading...</>
+  if (!loading) {
+    if (moderator) {
+      titleText = (
+        <>
+          You have the <b>moderator</b> role
+        </>
+      )
+      subtitleText = <>You can delete everyone's messages</>
+      Icon = ShieldCheckIcon
+      iconColour = 'text-orange-500'
+      iconBackground = 'bg-orange-100'
+    } else {
+      titleText = (
+        <>
+          You have the <b>participant</b> role
+        </>
+      )
+      subtitleText = <>You can delete your own messages</>
+      Icon = UserCircleIcon
+      iconColour = 'text-slate-800'
+      iconBackground = 'bg-slate-200'
```

Comment:
Rather than mutating all of these values, we should abstract out a component that takes these as props 👍